### PR TITLE
Adds /v0/profile/contacts, mocked VA Profile Health Benefit service

### PR DESCRIPTION
**Pull request by radavis**
## Summary

- Adds `GET /v0/profile/contacts`, returning Next-of-kin and Emergency Contact records for a Veteran
- Feature controlled with feature toggle: `profile_contacts`
- Adds `VAProfile::HealthBenefit::Service` and supporting code for a planned future integration. Responds with mock data

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#66602

## Testing done

- *Describe what the old behavior was prior to the change*
  - Prior behavior: `GET /v0/profile/contacts` returns 4xx.
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
  - To verify: Sign in and visit `/v0/profile/contacts`. Two mocked records will be returned.

## What areas of the site does it impact?

- Adds a `Contact` resource to the `/v0/profile` API namespace

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Comment by rmtolmach** [comment]
👋 I merged some rspec changes on Friday. I wanted to be sure your branch was up-to-date so it would catch any errors pre-merge, so that's why I merged master into your branch.

**Review by RachalCassity — DISMISSED** [review]

**Review by tpharrison — COMMENTED** [review]
Just two things we chatted about:
- Keep `vfs-authenticated-experience-backend` as codeowners, but also add your team
- Change `ContactInformationController` to `AssociatedPersonsController`.  This is just a suggestion based on what we may build going forward.

**Comment by radavis** [comment]
>     * Change `ContactInformationController` to `AssociatedPersonsController`.  This is just a suggestion based on what we may build going forward.

Hey @tpharrison, I have a `ContactsController` and an API path named `/v0/profile/contacts`. Does that pose any conflicts with what you anticipate adding to `/v0/profile`?

**Comment by tpharrison** [comment]
> > ```
> > * Change `ContactInformationController` to `AssociatedPersonsController`.  This is just a suggestion based on what we may build going forward.
> > ```
> 
> Hey @tpharrison, I have a `ContactsController` and an API path named `/v0/profile/contacts`. Does that pose any conflicts with what you anticipate adding to `/v0/profile`?

That works 👍

**Comment by radavis** [comment]
> 1 Error
> 🚫 	New entry on line 9 of CODEOWNERS does not include @department-of-veterans-affairs/backend-review-group.
> Please add @department-of-veterans-affairs/backend-review-group to the entry
> Offending line: `app/controllers/v0/profile/contacts_controller.rb @department-of-veterans-affairs/vfs-authenticated-experience-backend @department-of-veterans-affairs/vfs-health-modernization-initiative`
> 	1 Warning
> ⚠️ 	This PR changes `399` LoC (not counting whitespace/newlines).
> 
> Generated by 🚫 [Danger](https://danger.systems/)

I can change this, but _should_ **I** be changing this?

File: run
Comment (tpharrison) [inline run:1]: Should platform review this?

**Review by tpharrison — CHANGES_REQUESTED** [review]

File: run
Comment (radavis) [reply run:1]: Removed this file.

**Review by tpharrison — APPROVED** [review]

File: .github/CODEOWNERS
Comment (RachalCassity) [inline .github/CODEOWNERS:1]: Thank you for updating this file. :)

**Review by beaugrantham-va — APPROVED** [review]
Lighthouse is unaffected by this PR.

**Comment by radavis** [comment]
lighthouse-banana-peels team is being asked to review because of this line in `.github/CODEOWNERS`

```
# line: 37
app/controllers/v0/apidocs_controller.rb @department-of-veterans-affairs/lighthouse-banana-peels
```

Disability-Experience team is being asked to review because of this line in `.github/CODEOWNERS`

```
# line: 927
lib/va_profile @department-of-veterans-affairs/Disability-Experience @department-of-veterans-affairs/vfs-authenticated-experience-backend @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
```

**Review by RachalCassity — APPROVED** [review]

File: config/routes.rb [hunk 1]
```diff
@@ -373,6 +374,7 @@
     post 'terms_of_use_agreements/:version/accept', to: 'terms_of_use_agreements#accept'
     post 'terms_of_use_agreements/:version/decline', to: 'terms_of_use_agreements#decline'
   end
+  # end /v0
```
Comment (tommasina-va) [inline config/routes.rb:377]: I think we can probably remove this?

**Pull request merged**